doc/document.go

```diff
@@ -133,35 +136,31 @@ func (d Document) Equal(other Document) bool {
 	return true
 }
 
-// Validate validates the given document and returns its ID if it has one.
-func (d Document) Validate() ([]byte, error) {
+// Validate returns a bool indicating whether the document is valid.
+func (d Document) Validate() error {
```

Comment:
should this call `HasID()`? maybe add a test for this case too

Reply:
I don't think we need to require an ID here, in fact, in the segment we currently check if a document is valid before checking its ID. We might want to remove the check for the fields though in case we ever want to index just an ID. Not sure if that would ever be needed though.